src/main/java/com/miquido/validoctor/multirule/ReflexiveMultiRuleBuilder.java

```diff
@@ -126,13 +127,43 @@
     return this;
   }
 
+  /**
+   * Applies given MultiRules to all properties of given type found in the subject class.
+   * For use with complex types.
+   * Every public method returning a given type and starting with "get" is treated as a property getter.
+   *
+   * @param propertyClass    class of properties to apply the rules to
+   * @param strictClassMatch if true, only properties with the exact specified class will have the rules applied.
+   *                         If false, rules will be applied to properties of subtypes as well.
+   * @param rules            MultiRules to apply
+   * @param <PropertyType>   type of property
+   * @return builder for chaining
+   */
+  @SafeVarargs
+  public final <PropertyType> ReflexiveMultiRuleBuilder<PatientType> addRulesForAll(Class<? extends PropertyType> propertyClass,
```

Comment:
this one will not work for the Boolean getters `is`

Reply:
this is intentional. This method accepts MultiRules - their purpose is validation of complex objects. For validation of primitives, including booleans, overload found just below needs to be used. It accepts SimpleRules and checks for isSomething methods in case of boolean.